packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -414,14 +414,6 @@ Future<XcodeBuildResult> buildXcodeProject({
   }
   if (buildResult != null && buildResult.exitCode != 0) {
     globals.printStatus('Failed to build iOS app');
-    if (buildResult.stderr.isNotEmpty) {
```

Comment:
So the assumption is that the actual error was already echo'd by xcode_backend?

Reply:
Yes, that because the build failed there will be at least one error message from Xcode that will be parsed and already shown.  That includes any failure from `assemble`, including a dart compilation issue.